src/support/modify.c

```diff
@@ -425,6 +409,32 @@ __wt_modify_apply_item(WT_SESSION_IMPL *session, WT_ITEM *value, const void *mod
     return (0);
 }
 
+/*
+ * __wt_modify_apply_buf --
+ *     Apply a single set of WT_MODIFY changes to a buffer.
+ */
+int
+__wt_modify_apply_buf(WT_SESSION_IMPL *session, WT_ITEM *buf, const void *modify)
+{
+    return (__modify_apply_item(session, buf, modify, S2BT(session)->value_format[0] == 'S'));
+}
+
+/*
+ * __wt_modify_apply --
+ *     Apply a single set of WT_MODIFY changes to a cursor buffer.
+ */
+int
+__wt_modify_apply(WT_CURSOR *cursor, const void *modify)
+{
+    WT_SESSION_IMPL *session;
+    bool sformat;
+
+    session = (WT_SESSION_IMPL *)cursor->session;
+    sformat = cursor->value_format[0] == 'S';
+
+    return (__modify_apply_item(session, &cursor->value, modify, sformat));
+}
```

Comment:
It's odd that __wt_modify_apply and __wt_modify_apply_buf both do exactly the same thing but with different approaches (one uses a local variable). They should be consistent.

In fact, can't you fold the check into __wt_modify_apply_item, and remove the `sformat` parameter? Then __wt_modify_apply_buf can go away, and `__wt_modify_apply` becomes `WT_WITH_BTREE(cursor->btree, __wt_modify_apply_item(session, &cursor->value, modify));`

It's also possible that we don't need the WT_WITH_BTREE at all - I expect it'll match the btree in the cursor already.